[Jxleejiaxin] iP

Jxleejiaxin opened the pull request:
### DukePro for Pros (Free [DLC)] 

> "Life is too complicated not to be orderly." - Martha Stewart ([source](https://dansilvestre.com/productivity-quotes/))

DukePro for Pros empowers you to remain organized when things get too hectic. It's,

- text-based
- easy to learn
- FAST to use
- Comes with free DLC

All you need to do is,
1. download it from [here](https://github.com/Jxleejiaxin/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!** with **FREE** DLC

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
- [ ] DLC (Free) (coming soon)

Commit:
Add Gradle support

Jxleejiaxin committed:
Implemented Level 1: Greet,Echo,Exit

Jxleejiaxin committed:
Fixed error where inputs with space are echoed incorrectly

Jxleejiaxin committed:
Implement Add, List: ability to store text entered by user and display them when requested

Jxleejiaxin committed:
Add ability to mark task as done/undone

Jxleejiaxin committed:
Level-4: Add tasks types ToDos,Events,Deadlines

Jxleejiaxin committed:
Add horizontal lines in responses from Duke, EXPECTED.TXT and
input.txt for automated text UI testing

Jxleejiaxin committed:
Add minimal for Level 5 due to lack of time. Will improve upon in the
error handling in the future.

Jxleejiaxin committed:
Fixed bugs introduced by error handling due to copy pasting error
handling of ToDo tasks to Deadline tasks.

Jxleejiaxin committed:
Level-6: Add support for deleting tasks from the list with elementary
error handling

Jxleejiaxin committed:
Add Level-7: Save tasks to hard disk.

Jxleejiaxin committed:
Add Level-8: DateTime

Jxleejiaxin committed:
Merge branch 'branch-Level-7'

Jxleejiaxin committed:
Merge branch 'branch-Level-8'

evanpy commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    protected String from;
+    protected String to;
```
"from" and "to" might be ambiguous, well at least from my pov. Instead of time, I could think of it in terms of place.
Other than that, nice code 👍🏻 good to go.

evanpy reviewed (commented):
LGTM, almost. Just a few nits to fix. 😄

Jxleejiaxin committed:
Merge branch 'add-gradle-support'

Nafeij commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
This could be better encapsulated. Perhaps make it `public final`?

Jxleejiaxin replied:
I have made it private final. Thank you!

Nafeij commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,263 @@
+                    taskIndex = scanner.nextInt() - 1;
+                    currentTask = taskArrayList.get(taskIndex);
+                    currentTask.markAsDone();
+                    System.out.println("\n____________________________________________________________");
```
This could be better abstracted to fit OO principles. Perhaps a separate `Parser` / `UI` class?

Nafeij commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,263 @@
+        System.out.println("\nBye. Hope to see you again soon!\n");
+    }
+
+    //helper method that prints the reply to each added task
```
Could these method headers be made JavaDoc-compliant?

Nafeij commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by:" + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
This string operation is a bit confusing. Would you consider using `String.format(...` or splitting it into multiple lines?

Jxleejiaxin replied:
Good idea. Thanks for the comments!

Nafeij reviewed (commented):
Some methods could have been better abstracted or encapsulated. Otherwise good work!

Jxleejiaxin committed:
Extract Ui, Storage, Parser, TaskList and various Command classes

Jxleejiaxin committed:
Divide and organize classes into packages

Jxleejiaxin committed:
Add two basic JUnit tests for DeleteCommand and AddDeadlineCommand

Jxleejiaxin committed:
Change build.gradle to reflect change in Duke package

Jxleejiaxin committed:
Minor changes in field access

Jxleejiaxin committed:
Add JavaDocs for classes and methods

Jxleejiaxin committed:
Commit minor changes to code

Jxleejiaxin committed:
Fix minor error in Parser class when refactoring code

Jxleejiaxin committed:
Add Level 9 Find Command, abstracted showIndexedList method to Ui as
well

Jxleejiaxin committed:
Merge branch 'branch-A-JavaDoc'

Jxleejiaxin committed:
Merge branch 'branch-Level-9'

Jxleejiaxin committed:
Fix minor bug where mark command only accepted 1

Jxleejiaxin committed:
Update README and edit some syntax for readability

CShuwen commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    /**
+     * Deletes task from TaskList by index.
+     * Prompts Ui to display deleted task followed by reminder of number of tasks in TaskList.
+     * Prompts Storage to overwrite existing task storage file with new TaskList.
```
I like how detailed the descriptions are but the summary of the method may need to be written in one sentence only.

CShuwen commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,23 @@
+
+    /**
+     * Determine if Command is an instance of ExitCommand
+     * @return true if Command subclass is ExitCommand
```
A minor point: In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)
Maybe it would be better to change the start of the sentence to "Returns boolean value of......."

CShuwen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+public class Duke {
```
Do you think it would be better to define the class Duke despite its functionality was already mentioned in the README?
Was there any reason why description of class was not added to Duke?

CShuwen commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,38 @@
+        }
+    }
+
+
```
I would suggest to remove excessive lines. Can try out using CheckStyle to detect coding style violations if you are interested. 😸

CShuwen reviewed (commented):
Overall, I found your code easy to read and your javaDocs are well written. 
I like the way you define your classes, it keeps each method short and makes them flexible to new changes.
Just take note of some coding standard violations. I have commented on a few that I spotted. Good job! 👍

Jxleejiaxin committed:
Add checkstyle to detect coding style violations

Jxleejiaxin committed:
Add CheckStyle to detect coding style violations

Jxleejiaxin committed:
Merge branch 'master' of https://github.com/Jxleejiaxin/ip

Jxleejiaxin committed:
Add javaFx part 2

Jxleejiaxin committed:
Add basics of GUI

Jxleejiaxin committed:
Add assertions

Jxleejiaxin committed:
Merge pull request #2 from Jxleejiaxin/branch-A-Assertions

Add assertions

Jxleejiaxin committed:
Refactor Ui class to Formatter class

Ui class is not required when GUI is handled through JavaFx.
Formatting is still required to get the desired output.

The formatting of strings is encapsulated in a more suitable class.
Desired output can be achieved by editing Formatter class.

Jxleejiaxin committed:
Extract markIfDone function in Parser for code quality

Jxleejiaxin committed:
Fix checkstyle errors

Jxleejiaxin committed:
Merge pull request #3 from Jxleejiaxin/branch-A-CodeQuality

Improve Code Quality

Jxleejiaxin committed:
Add SetSourceCommand to change data source of Duke

SetSourceCommand is created in Duke by inputting "source [path to
file]". Users can switch between data sources and Duke will
automatically load the existing tasks in the data sources.

Jxleejiaxin committed:
Fix checkstyle errors

Jxleejiaxin committed:
Add isValidAddEventCommand method in Parser class

Removes an extra try-catch statement in the long parseFromUser method.
Avoids arrowhead style code.

Jxleejiaxin committed:
Improve GUI of Duke

Add different profile picture for Duke and User. Add borders to
text field. Add "command: " prefix to each user input to match the
assymetrical nature of the conversation by a little.

Jxleejiaxin committed:
Add personality of a chat(BOT)

Jxleejiaxin committed:
Add user guide and representative screenshot

Jxleejiaxin committed:
Change build.gradle file to make Launcher the application

Jxleejiaxin committed:
Update user guide
